Packs/ShadowxSOCAI/Integrations/ShadowxSOCAI/ShadowxSOCAI.py

```diff
@@ -0,0 +1,535 @@
+    ai_driver_id = demisto.params().get('ai_driver_id') or ''
+    default_policy_id = demisto.params().get('policy_id') or ''
+
+    log = args.get('log')
+    ip_addr = args.get('ip_addr')
+    user_name = args.get('user_name')
+    policy_id = args.get('policy_id') or default_policy_id
+
+    security_log = log if not ip_addr else f'{log} [ip:{ip_addr}]'
+
+    payload = {
+        # API/FreeText shapes (lenient)
+        "SearchText": log,
+        "IpAddr": ip_addr,
+        "UserName": user_name,
+        "PolicyId": policy_id,
+        # SecurityTasks create shape
+        "TaskName": task_name,
+        "AssignedUserID": assigned_user_id,
+        "AIDriverID": ai_driver_id,
+        "PolicyID": policy_id,
+        "Subject": user_name or "",
+        "SecurityLog": security_log,
+        "Status": 1
+    }
+    return payload
+
+
+def test_module():
+    """If API key set → simple GET / with Bearer; else try cookie login."""
+    base_url = (demisto.params().get('url') or '').rstrip('/')
+    api_key = demisto.params().get('api_key') or ''
+    email = demisto.params().get('user_email') or ''
+    password = demisto.params().get('user_password') or ''
+
+    _require_param('url', base_url)
+    s = _new_session()
+
+    if api_key:
+        r = s.get(f'{base_url}/', headers=api_key_headers(api_key), verify=_verify_flag(), timeout=15)
+        if r.status_code in (200, 302):
+            return_results('ok')
+        else:
+            return_error(f'API key test failed: {r.status_code} {_debug_slice(r.text)}')
+    else:
+        _require_param('user_email', email)
+        _require_param('user_password', password)
+        _, _ = login_to_dotnet_api(s, base_url, email, password)  # raises on failure
+        return_results('ok')
+
+
+def shadowx_submit_task_command():
+    args = demisto.args()
+    base_url = (demisto.params().get('url') or '').rstrip('/')
+    api_key = demisto.params().get('api_key') or ''
+
+    # Required arg
+    log = args.get('log')
+    if not log:
+        return_error("'log' is required.")
+
+    payload = _build_task_payload_from_args(args)
+    wait_seconds = int(args.get('wait_seconds', 0))
+
+    s = _new_session()
+
+    if api_key:
+        # API mode (Bearer)
+        submit_res = submit_with_api_key(s, base_url, api_key, payload)
+        if submit_res.status_code != 200:
+            return_error(f"Task submit failed (API key): {submit_res.status_code} {_debug_slice(submit_res.text, 600)}")
+
+        submit_json: Dict[str, Any] = {}
+        try:
+            submit_json = submit_res.json()
+        except Exception:
+            pass
+
+        task_id = _extract_task_id(submit_json) or ""
+        ui_url = f"{base_url.rstrip('/')}/SecurityTasks/Details?taskID={task_id}" if task_id else ""
+
+        # Optional polling (if TaskId exists)
+        if wait_seconds > 0 and task_id:
+            end = time.time() + wait_seconds
+            while time.time() < end:
+                time.sleep(3)
```

Comment:
The polling mechanism implemented with `wait_seconds` has two issues: the use of `time.sleep()` is discouraged, and the polling logic for cookie mode is incomplete.